src/DIRAC/ConfigurationSystem/Client/Utilities.py

```diff
@@ -578,3 +569,46 @@ def getDIRACGOCDictionary():
 
   log.debug('End function.')
   return S_OK(dictionary)
+
+
+def getAuthAPI():
+  """ Get Auth REST API url
+
+      :return: str
+  """
+  return gConfig.getValue("/Systems/Framework/%s/URLs/AuthAPI" % getSystemInstance("Framework"))
+
+
+def getAuthorisationServerMetadata(issuer=None):
+  """ Get authoraisation server metadata
+
+      :return: S_OK(dict)/S_ERROR()
+  """
+  data = {'issuer': issuer}
+
+  result = gConfig.getSections('/DIRAC')
+  if result['OK']:
+    if 'Authorization' in result['Value']:
+      result = gConfig.getOptionsDictRecursively('/DIRAC/Authorization')
+      if result['OK']:
+        data.update(result['Value'])
+  if not result['OK']:
+    return result
+  if not data['issuer']:
+    data['issuer'] = getAuthAPI()
+  if not data['issuer']:
+    return S_ERROR('No issuer found in DIRAC authorization server configuration.')
+
+  # Search values with type list
+  for key, v in data.items():
+    data[key] = [e for e in v.replace(', ', ',').split(',') if e] if ',' in v else v
```

Comment:
You can use a dict comprehension.

Reply:
and use meaningful name or at least explain what they are supposed to be